drizzlepac/hapsequencer.py

```diff
@@ -1274,3 +1277,33 @@ def delete_ramp_exposures(obj_list, type_of_list):
         obj_list.append(temp.pop())
 
     return ramp_filenames_list
+
+
+def _verify_sci_hdrname(filename):
+    """Insures that HDRNAME keyword is populated in SCI extensions.
+
+    This function checks to make sure the HDRNAME keyword in the SCI
+    extension of the science image `filename` is populated with a valid
+    non-empty string.
+    """
+    fhdu, closefits = proc_utils._process_input(filename)
+
+    # Find all extensions to be updated
+    numext = countExtn(fhdu, extname='SCI')
+
+    for ext in range(1, numext + 1):
+        sciext = ('sci', ext)
+        scihdr = fhdu[sciext].header
+        if 'hdrname' not in scihdr or scihdr['hdrname'] == '':
+            # We need to create a valid value for the keyword
```

Comment:
This is going to seem paranoid, but should we make sure it is not a blank string too?

Reply:
The logic in the 'if' statement looks to see whether the current value is blank (''), and if so, initiates replacing that value.  In addition, the new value being generated will never be blank since it is being generated from the filename itself and the WCSNAME keyword value (which is never going to be blank at this point in the processing).  So, I believe this code adequately protects against blank strings in both the input and output values of this keyword.

Reply:
Sorry, I am being dense.  I meant contains a blank versus being empty (' ' vs '').

Reply:
You are not the only one being dense.  That actually makes sense, and would be a simple change.  It will be included in the next commit.

Reply:
Fixed in 16785770